internal/service/advancedcluster/plan_modifier.go

```
@@ -178,17 +183,62 @@ func autoScalingUsed(ctx context.Context, diags *diag.Diagnostics, state, plan *
 					if autoscaling == nil {
 						continue
 					}
-					if autoscaling.ComputeEnabled.ValueBool() {
-						computedUsed = true
-					}
-					if autoscaling.DiskGBEnabled.ValueBool() {
-						diskUsed = true
+					if autoscaling.ComputeEnabled.ValueBool() || autoscaling.DiskGBEnabled.ValueBool() {
+						return true
 					}
 				}
 			}
 		}
 	}
-	return
+	return false
+}
+
+// isReadOnlySpecsDeleted detects if any read_only_specs block with node_count > 0 was deleted from the plan.
+func isReadOnlySpecsDeleted(ctx context.Context, diags *diag.Diagnostics, state, plan *TFModel) bool {
+	stateRepSpecsTF := TFModelList[TFReplicationSpecsModel](ctx, diags, state.ReplicationSpecs)
+	planRepSpecsTF := TFModelList[TFReplicationSpecsModel](ctx, diags, plan.ReplicationSpecs)
+	if diags.HasError() {
+		return false
+	}
+	for i := range minLen(planRepSpecsTF, stateRepSpecsTF) {
+		stateRegionConfigsTF := TFModelList[TFRegionConfigsModel](ctx, diags, stateRepSpecsTF[i].RegionConfigs)
+		planRegionConfigsTF := TFModelList[TFRegionConfigsModel](ctx, diags, planRepSpecsTF[i].RegionConfigs)
+		if diags.HasError() {
+			return false
+		}
```

very unlikely, but should we return these errors if encountered?

---

I don't think so, similarly to adjustRegionConfigsChildren, if we find some errors here is probably because cluster topology changed so we won't do the optimization but it's not a "real error" to surface to the user